org.lflang/src/org/lflang/cli/Lfc.java

```diff
@@ -152,57 +163,53 @@ public static void main(Io io, final String... args) {
         cliMain("lfc", Lfc.class, io, args);
     }
 
-    @Override
-    protected Options getOptions() {
-        return CLIOption.getOptions();
-    }
-
     /**
      * Load the resource, validate it, and, invoke the code generator.
      */
     @Override
-    protected void runTool(CommandLine cmd, List<Path> files) {
-        Properties properties = this.filterProps(cmd, CLIOption.getPassedOptions());
-        String pathOption = CLIOption.OUTPUT_PATH.option.getOpt();
-        Path root = null;
-        if (cmd.hasOption(pathOption)) {
-            root = io.getWd().resolve(cmd.getOptionValue(pathOption)).normalize();
-            if (!Files.exists(root)) { // FIXME: Create it instead?
-                reporter.printFatalErrorAndExit("Output location '" + root + "' does not exist.");
-            }
-            if (!Files.isDirectory(root)) {
-                reporter.printFatalErrorAndExit(
-                    "Output location '" + root + "' is not a directory.");
-            }
+    public void run() {
+        List<Path> paths = getInputPaths();
+        final Path outputRoot = getOutputRoot();
+        // Hard code the props based on the options we want.
+        Properties properties = this.filterPassOnProps();
+
+        try {
+            // Invoke the generator on all input file paths.
+            invokeGenerator(paths, outputRoot, properties);
+        } catch (RuntimeException e) {
+            reporter.printFatalErrorAndExit("An unexpected error occurred:", e);
         }
+    }
 
-        for (Path path : files) {
-            if (!Files.exists(path)) {
-                reporter.printFatalErrorAndExit(path + ": No such file or directory");
-            }
-        }
+    /**
+     * Invoke the code generator on the given validated file paths.
+     */
+    private void invokeGenerator(
+            List<Path> files, Path root, Properties properties) {
         for (Path path : files) {
             path = toAbsolutePath(path);
             String outputPath = getActualOutputPath(root, path).toString();
             this.fileAccess.setOutputPath(outputPath);
 
             final Resource resource = getResource(path);
             if (resource == null) {
-                reporter.printFatalErrorAndExit(
-                    path + " is not an LF file. Use the .lf file extension to denote LF files.");
-            }
-            else if (cmd.hasOption(CLIOption.FEDERATED.option.getOpt())) {
+                reporter.printFatalErrorAndExit(path 
+                    + " is not an LF file. Use the .lf file extension to"
+                    + " denote LF files.");
+            } else if (federated) {
                 if (!ASTUtils.makeFederated(resource)) {
```

Comment:
It looks wrong to me to invoke a AST transformation at this point. This should really happen in the generators. This is not an issue with this PR though, as the code already existed before.